client/state/utils.js

```diff
@@ -205,9 +203,9 @@ export function extendAction( action, data ) {
 		return merge( {}, action, data );
 	}
 
-	return dispatch => {
-		const newDispatch = flow( partialRight( extendAction, data ), dispatch );
-		return action( newDispatch );
+	return ( dispatch, getState ) => {
+		const newDispatch = a => dispatch( extendAction( a, data ) );
```

Comment:
FFTI since it'll be hard to fix, but I don't love the single character variable name. The action parameter higher up is a lie when its a fn.

Reply:
I tried to think of a better parameter name, but didn't come up with anything. I'll leave it as `a` -- it shouldn't be very hard to figure out that it's an action.

The `action` parameter higher up can be either object or an action thunk -- and both deserve the name `action`, because they are dispatchable. But your mileage may vary -- thunks are a bit controversial after all.